pg_search/src/postgres/customscan/pdbscan/exec_methods/fast_fields/mod.rs

```diff
@@ -405,3 +425,120 @@ pub fn estimate_cardinality(indexrel: &PgRelation, field: &str) -> Option<usize>
             .num_terms(),
     )
 }
+
+/// Process attributes using fast fields, creating a mapping and populating the datum array.
+/// This function is shared between the string and numeric fast field implementations.
+#[allow(clippy::too_many_arguments)]
+pub unsafe fn process_fast_fields(
```

Comment:
Ditto `process` being a bit too generic of a name. I don't understand enough about this function to give it a better name, but you probably do after having extracted it.

Reply:
Good suggestion. Renamed to `extract_data_from_fast_fields`.